q2_fmt/_peds.py

```diff
@@ -0,0 +1,147 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2022-2022, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import qiime2
+import pandas as pd
+
+
+def sample_peds(table: pd.DataFrame, metadata: qiime2.Metadata,
+                time_column: str, reference_column: str, subject_column: str,
+                filter_missing_references: bool = False,
+                drop_incomplete_subjects: bool = False) -> (pd.DataFrame):
+    # TODO: Make incomplete samples possible move this to heatmap
+    metadata = metadata.to_dataframe()
+
+    try:
+        num_timepoints = metadata[time_column].dropna().unique().size
+    except Exception:
+        raise KeyError('There was and error finding %s in the metadata'
+                       % time_column)
+    try:
+        metadata[time_column].dropna().astype(int)
+    except Exception:
+        raise KeyError('%s must be numeric' % time_column)
+    try:
+        subject_occcurance_df = (metadata[subject_column].value_counts()
+                                 .to_frame())
+    except Exception:
+        raise KeyError('There was and error finding %s in the metadata'
```

Comment:
Is there any way we can make what we're excepting a bit more specific (goes for anywhere it just says `except Exception`)? The errors you reraise seem to be related to specific failure states so unless we can be really confident that's the only thing that could raise an exception in that line of code (it probably isn't) we should probably make sure the original exception is what we think it is. Also, you can do something like this
 
```
except Exception as e:
    raise KeyError('Oh no bad') from e
```
And now your Keyerror will include information about the other exception in the traceback which can be helpful sometimes.